src/msg/async/AsyncConnection.cc

```diff
@@ -2240,9 +2240,8 @@ ssize_t AsyncConnection::write_message(Message *m, bufferlist& bl, bool more)
                              << " off " << header.data_off << dendl;
 
   if ((bl.length() <= ASYNC_COALESCE_THRESHOLD) && (bl.buffers().size() > 1)) {
-    std::list<buffer::ptr>::const_iterator pb;
-    for (pb = bl.buffers().begin(); pb != bl.buffers().end(); ++pb) {
-      outcoming_bl.append((char*)pb->c_str(), pb->length());
+    for (auto &pb:bl.buffers()) {
```

Comment:
@joscollin because there is a temp object constructor happening per loop if write before. but there is only one time invoking bl.buffers().end() for the range-base for statement. so I think it's good for code and performance especially IO path. thanks.